src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php

```diff
@@ -32,7 +33,21 @@ public function isFresh($timestamp)
             return false;
         }
 
-        return @filemtime($this->filePath) <= $timestamp;
+        // has the file *not* been modified? Definitely fresh
+        if (@filemtime($this->filePath) <= $timestamp) {
+            return true;
+        }
+
+        try {
+            $reflectionClass = new \ReflectionClass($this->class);
+        } catch (\ReflectionException $e) {
+            // the class does not exist anymore!
+            return false;
+        }
+
+        $newSelf = AutowirePass::createResourceForClass($reflectionClass);
+
+        return $this == $newSelf;
```

Comment:
I thought about inlining but this was the original style (ref https://github.com/symfony/symfony/pull/18144/commits/49271e86f999ad84bb4f8787b844973f7c22f4fc#diff-04e2904117929f6ffe784d5be3cb1103L49).

So I'm fine either way. Final call ?

Doesn't a strict comparison always return true ? We need to compare their data, right ?